### PR TITLE
Add pdfmm

### PR DESCRIPTION
**Pull request by xmnlab**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xmnlab** [comment]
@conda-forge/help-c-cpp this PR is ready for review! thanks!

File: recipes/pdfmm/build.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# configure cmake
+rm -rf build
+cmake -B build \
+  -DCMAKE_PREFIX_PATH=${PREFIX} \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_CXX_STANDARD=17 \
+  -DPDFMM_BUILD_STATIC=OFF \
+  -DPDFMM_USE_VISIBILITY=ON \
+  -DPDFMM_BUILD_LIB_ONLY=ON
+
```
Comment (carterbox) [inline recipes/pdfmm/build.sh:15]:
```suggestion
cmake -B build \
  -DCMAKE_PREFIX_PATH=${PREFIX} \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_CXX_STANDARD=17 \
  -DPDFMM_BUILD_STATIC=OFF \
  -DPDFMM_USE_VISIBILITY=ON \
  -DPDFMM_BUILD_LIB_ONLY=ON \
  ${CMAKE_ARGS}

```
We have an environment variable that automatically sets conda-forge specific args for cmake.


https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=cmake_args#using-cmake

File: recipes/pdfmm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - fontconfig
+    - openssl
+    - zlib
+    - libjpeg-turbo
+    - libtiff
+    - libpng
+    - libidn11
+    - libxml2
+  run:
+    - libidn11
```
Comment (carterbox) [inline recipes/pdfmm/meta.yaml:33]:
```suggestion
    - {{ pin_compatible('libidn11') }}
```

File: recipes/pdfmm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libpdfmm.so  # [linux]
+
```
Comment (carterbox) [inline recipes/pdfmm/meta.yaml:38]:
```suggestion
test:
  commands:
    - test -f ${PREFIX}/lib/libpdfmm.so
    - test -f ${PREFIX}/lib/libpdfmm.so.{{ version }}
    - test -f ${PREFIX}/include/pdfmm/pdfmm.h
    - test -f ${PREFIX}/lib/pkgconfig/libpdfmm.pc
```

File: recipes/pdfmm/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pdfmm/pdfmm/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 5883f4f1e01fb657f6b43a25115640292458c2b9db3f87d4517d2e341019a590
+
+build:
+  number: 0
+  skip: true  # [not linux]
```
Comment (carterbox) [inline recipes/pdfmm/meta.yaml:14]:
```suggestion
build:
  number: 0
  skip: true  # [not linux]
  # TODO: Reassess pinning at version 1.0
  run_exports:
    - {{ pin_subpackage(name|lower, max_pin='x.x.x') }}
```
Run export with max_pin according to *ABI* compatability. Might be safe to pin to 'x.x' if there will be no ABI breaks except at MINOR versions.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/pdfmm/build.sh (hunk 1 above)
Comment (xmnlab) [reply recipes/pdfmm/build.sh:15]: I didn't know about that! thank you for the explanation!

File: recipes/pdfmm/meta.yaml (hunk 4 above)
Comment (xmnlab) [reply recipes/pdfmm/meta.yaml:14]: thanks for the explanation! I have package for a long time here .. but `run_exports` is still something that I never understand well XD I will spend some time to understand it better! thank you so much!

**Comment by xmnlab** [comment]
@carterbox thank you so much for your review. I applied your suggestions and also fixed the license identifier as well.
thank you!

**Comment by xmnlab** [comment]
@carterbox @conda-forge/help-c-cpp just a friendly reminder about this PR. thanks!

**Review by carterbox — APPROVED** [review]

**Pull request merged**